src/posix/platform/netif.cpp

```diff
@@ -1167,7 +1175,10 @@ static void processNetifLinkEvent(otInstance *aInstance, struct nlmsghdr *aNetli
     if (isUp && gNat64Cidr.mLength > 0)
     {
         SuccessOrExit(error = otNat64SetIp4Cidr(gInstance, &gNat64Cidr));
-        AddIp4Route(gNat64Cidr, kNat64RoutePriority);
+        if (otNat64GetTranslatorState(gInstance) == OT_NAT64_STATE_ACTIVE)
+        {
+            AddIp4Route(gNat64Cidr, kNat64RoutePriority);
+        }
```

Comment:
We can skip this since otNat64SetIp4Cidr should trigger the state change callback.